### PR TITLE
fix new warning in nightly

### PR DESCRIPTION
**Pull request by gilescope**
There's one place where we are intentionally doing this so have put an exclude in for that.

Same as https://github.com/paritytech/cumulus/pull/2721

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2962679

**Comment by gilescope** [comment]
Hmm, `#[allow(hidden_glob_reexports)]` is not yet understood by the CI's version of rust.
For now let's remove all but this one warning.

**Review by ggwpez — APPROVED** [review]

File: frame/staking/src/lib.rs [hunk 1]
```diff
@@ -297,6 +297,7 @@ pub mod migrations;
 pub mod slashing;
 pub mod weights;
 
+// TODO: set #[allow(hidden_glob_reexports)] once stable understands this term.
```
[the remark this answers is not in the document]
Comment (gilescope) [reply frame/staking/src/lib.rs:300]: The CI will error as soon as it's upgraded to a stable that includes this warning so we won't miss it.

**Review by gui1117 — APPROVED** [review]

File: frame/staking/src/lib.rs (hunk 1 above)
Comment (bkchr) [inline frame/staking/src/lib.rs:300]: Can we not just solve this instead of adding this?

Comment (gilescope) [reply frame/staking/src/lib.rs]: Good point - the explicit imports aren't too bad.

**Review by bkchr — APPROVED** [review]

**Pull request merged**